k8sdeps/transformer/target/target.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package target
+
+import (
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/kustomize/v3/pkg/gvk"
+	"sigs.k8s.io/kustomize/v3/pkg/resource"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+)
+
+// FindPatchTargets returns a list of resources that
+// are pointed by targets
+func FindPatchTargets(targets types.Targets,
```

Comment:
Want to keep k8sdeps small.

This is a search though resources, so please put it in ResMap, i.e.
```
type ResMap interface {
  Select(types.Selector) []*resource.Resource
}
```

Also, lets change the name of `Targets` to `Selector`.  :)

It makes more sense, given its members, given that its one thing, not a slice of things, etc.

Reply:
Added this function to the `ResMap` interface.